src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.model;
+
+public class FruitTransaction extends Fruit {
+    private Operation operation;
+
+    public FruitTransaction(String operation, String fruit, int quantity) {
+        super(fruit,quantity);
+        switch (operation) {
+            case "b" -> this.operation = Operation.BALANCE;
+            case "s" -> this.operation = Operation.SUPPLY;
+            case "p" -> this.operation = Operation.PURCHASE;
+            case "r" -> this.operation = Operation.RETURN;
+            default -> throw new RuntimeException("Can't find operation");
```

Comment:
The exception message 'Can't find operation' could be more informative. Consider including the invalid operation code in the message to aid debugging, e.g., 'Invalid operation code: ' + operation.